src/librbd/ImageWatcher.cc

```diff
@@ -184,6 +184,18 @@ int ImageWatcher::try_lock() {
       }
     }
 
+    if (m_image_ctx.cct->_conf->rbd_blacklist_on_break_lock) {
+      ldout(m_image_ctx.cct, 1) << "blacklisting client: " << locker << "@"
+				<< locker_address << dendl;
+      librados::Rados rados(m_image_ctx.md_ctx);
+      r = rados.blacklist_add(locker_address, 0);
```

Comment:
we should probably make the expiration configurable too, since the default on the monitor isn't, and if it was the global default might not make sense for rbd